nomad/state/events_state_store_test.go

```diff
@@ -0,0 +1,76 @@
+package state
+
+import (
+	"testing"
+	"time"
+
+	memdb "github.com/hashicorp/go-memdb"
+	"github.com/hashicorp/nomad/nomad/mock"
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/stretchr/testify/require"
+)
+
+func TestStateStore_AddSingleNodeEvent(t *testing.T) {
```

Comment:
Look how the other tests access the field that is being modified first, then modify and then assert the watch has been fired. This verifies that the update to the object will unblock blocked queries.